### DISCUSSION TITLE
Using this with express-session

### DISCUSSION THREAD

**Discussion by Chaphasilor** [porsager/postgres#239] [Q&A] [answered]
I have an express app that stores session data in the database. It uses [`express-session`](https://expressjs.com/en/resources/middleware/session.html) in the background as an abstraction layer, which in turn uses a [*session store*](https://expressjs.com/en/resources/middleware/session.html#session-store-implementation) to manage the sessions for various DBs.  
There's a Postgresql version called [`connect-pg-simple`](https://www.npmjs.com/package/connect-pg-simple), but it's only compatible with `pg` (`node-postgres`), or rather it [requires compatibility with `pg`'s `Pool` class](https://www.npmjs.com/package/connect-pg-simple#connection-options).

Given that I don't want to use `pg`, I was wondering if there already is a compatible session store that's compatible with `postgres` (I couldn't find one), or if there would be a way to "emulate" the `Pool` class using `postgres`.  

Otherwise, I see no other way than to fork `connect-pg-simple` and add support for `postgres` myself, but that would be a lot of work, so I'm looking for a better alternative :)

So **TL;DR:**  
Is there an Express session store compatible with `postgres` or is there a way to make `postgres` compatible with it (e.g. by adding a small abstraction/translation layer?

**Comment by Minigugus** [comment, marked as answer]
> Is there an Express session store compatible with postgres or is there a way to make postgres compatible with it (e.g. by adding a small abstraction/translation layer?

Is https://github.com/Minigugus/connect-postgres-simple what you are looking for? It's a basic rewrite of `connect-pg-simple` but using `postgres`, as you suggested. You can test it with `npm i express-session connect-postgres-simple`. The public API is almost the same as `connect-pg-simple` (all config properties are the same expect that thoses from `pg` where removed).

EDIT: `connect-postgres-simple` is now on [NPM](https://www.npmjs.com/package/connect-postgres-simple)

**Reply by Chaphasilor** [reply]
That''s *exactly* what I was looking for, thanks for reaching out! I'll try it out tomorrow :)

One question, did you test this with `postgres` 2.0? It should be compatible, just want to make sure \^\^

**Reply by Chaphasilor** [reply]
Just took a look at the `package.json` file...

> "​postgres​"​: ​"​^2.0.0-beta.10​"​,

Guess that answers my question :D

**Reply by Minigugus** [reply]
`connect-postgres-simple` is now on [NPM](https://www.npmjs.com/package/connect-postgres-simple) :+1:

This package was created yesterday, It probably contains bugs. Feel free to report them at https://github.com/Minigugus/connect-postgres-simple/issues :wink:
